config/battlefields.json

```diff
@@ -13,7 +13,7 @@
 		"bonuses": [
 			{
 				"type" : "MAGIC_SCHOOL_SKILL",
-				"subtype" : 0,
+				"subtype" : "spellSchool.any",
```

Comment:
Change is OK, but might cause hidden changes in mods, since subtype now use different underlying values. Or will your bonus conversion handle this?

Reply:
It is not, and I do not know how it should be handled by converter. Yes, this will require change inside mods (surely in WoG, unsure about others).

Reply:
Just thinking on how to detect such cases in mods (to help modders with update)
- require subtypes for *all* bonuses to be string, e.g. on validation level (will need more changes, e.g. damage types, movement types, probably others)
- require subtypes for *some* bonuses to be string, including this one. Not sure how to define this though. Perhaps as property of a bonus, inside CBonusType class? Or simply hardcoded list of bonuses that require strings (or opposite - that require integers)
- rename bonus and handle as bonus conversion

Reply:
I do not think than this should be handled in this PR

Reply:
okay, as long as it is handled in some way by 1.3

Reply:
I do not know how to handle it properly for now. Maybe we need some lists of bonuses which require a subtype with chosen identifier?

Reply:
Yeah, something like that.
We can add:
- mapping bonus type -> expected identifiers "class" for subtype value, e.g. `{ MAGIC_SCHOOL_SKILL, "spellSchool" }`
- when attempting to resolve identifier during load (e.g. resolveIdentifer/getIdentifier calls) we should provide this mapped bonus type. This will force identifier class check during bonus resolution
- would also be great to detect whether bonus uses subtypes to begin with - to provide modder with warning if he attempts to give subtype to a bonus that does not uses them

Not sure whether this is best approach or whether this will actually map well on our system, so not insisting on this particular form.